disk/block_test.go

```diff
@@ -0,0 +1,151 @@
+package disk
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"os"
+	"strings"
+	"testing"
+)
+
+const (
+	tmpTestFile = "/tmp/cfs_disk_block_test"
+)
+
+var testFilePattern = []byte{'a', 'b', 'c', 'd', 'e', 'f', 'g'}
+
+func setUpTestFile(length int64, t *testing.T) *os.File {
+	f, err := os.OpenFile(tmpTestFile, os.O_CREATE|os.O_TRUNC|os.O_RDWR, 0600)
+	if err != nil {
+		t.Fatalf("can not open test file: %v", err)
+		return nil
+	}
+
+	// fill tmp test file with test pattern
+	for length > 0 {
+		if length < int64(len(testFilePattern)) {
+			_, err = f.Write(testFilePattern[0 : length-1])
+		} else {
+			_, err = f.Write(testFilePattern)
+		}
+		if err != nil {
+			t.Fatalf("can not write to test file: %v", err)
+		}
+		length = length - int64(len(testFilePattern))
+	}
+
+	if err = f.Sync(); nil != err {
+		t.Fatalf("can not sync test file: %v", err)
+	}
+	return f
+}
+
+func TestPartlyReadBlock(t *testing.T) {
```

Comment:
well... it is a partial read, but the returned error should be a crc mismatch at least. io.eof is not an error, it just indicates the eof. so does this reveal a bug in our real code?